test/evaluate_integration_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2024 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	clarify "github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+	"github.com/clarify/clarify-go/views"
+)
+
+func TestEvaluate(t *testing.T) {
+	ctx := context.Background()
+	creds := getCredentials(t)
+	client := creds.Client(ctx)
+	prefix := createPrefix()
+	a := TestArgs{
+		ctx:         ctx,
+		integration: creds.Integration,
+		client:      client,
+		prefix:      prefix,
+	}
+
+	applyTestArgs(a, onlyError(insertDefault), onlyError(saveSignalsDefault), onlyError(publishSignalsDefault))
+
+	t0, t1 := getDefaultTimeRange()
+	ir, err := selectItemsDefault(a)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+
+	itemIDs := getListFromResponse(ir)
+
+	type testCase struct {
+		testArgs         TestArgs
+		itemIDs          []string
+		query            fields.ResourceQuery
+		data             fields.DataQuery
+		timeAggregation  fields.TimeAggregation
+		groupAggregation fields.GroupAggregation
+		expectedFields   func(*clarify.EvaluateResult) bool
+	}
+
+	test := func(tc testCase) func(t *testing.T) {
+		return func(t *testing.T) {
+			t.Helper()
+
+			result, err := evaluate(tc.testArgs.ctx, tc.testArgs.client, tc.itemIDs, tc.query, tc.data, tc.timeAggregation, tc.groupAggregation)
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			} else if !tc.expectedFields(result) {
+				t.Errorf("bugs!")
+			}
+
+			jsonEncode(t, result)
+		}
+	}
+
+	ef := func(er *clarify.EvaluateResult) bool {
+		// this could be done better!
+		// like some kind of order of items, groups, and calculations as test input,
+		// looped through here to verify they exist in the output.
+		keys := []string{"i0", "i1", "c1", "c2", "c3", "g1"}
+		ok := true
+		for _, key := range keys {
+			_, kok := er.Data[key]
+			ok = ok && kok
+		}
+
+		return ok
+	}
+
+	t.Run("basic evaluate test", test(testCase{
+		testArgs:         a,
+		itemIDs:          itemIDs,
+		query:            createAnnotationQuery(a.prefix),
+		data:             fields.Data().Where(fields.TimeRange(t0, t1)).RollupDuration(time.Hour, time.Monday),
+		timeAggregation:  fields.TimeAggregationAvg,
+		groupAggregation: fields.GroupAggregationAvg,
+		expectedFields:   ef,
+	}))
+
+	for ta := fields.TimeAggregationDefault + 1; ta <= fields.TimeAggregationRate; ta++ {
```

Comment:
nit: Maybe more clear to loop through an explicit slice of aggregates using range.